katharsis-core/src/main/java/io/katharsis/resource/registry/responseRepository/ResourceRepositoryAdapter.java

```diff
@@ -1,109 +1,189 @@
+				queryAdapter, id);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public JsonApiResponse findAll(QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				Object resources;
+				if (isAnnotated) {
+					resources = ((AnnotatedResourceRepositoryAdapter) resourceRepository).findAll(queryAdapter);
+				}
+				else if (resourceRepository instanceof QuerySpecResourceRepository) {
+					resources = ((QuerySpecResourceRepository) resourceRepository)
+							.findAll(request.getQuerySpec(resourceInformation.getResourceClass()));
+				}
+				else {
+					resources = ((ResourceRepository) resourceRepository).findAll(request.getQueryParams());
+				}
+				return getResponse(resourceRepository, resources, request);
+			}
+
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindAll(moduleRegistry.getResourceRegistry(),
+				queryAdapter);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public JsonApiResponse findAll(Iterable ids, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				Iterable<?> ids = request.getIds();
+				Object resources;
+				if (isAnnotated) {
+					resources = ((AnnotatedResourceRepositoryAdapter) resourceRepository).findAll(ids, queryAdapter);
+				}
+				else if (resourceRepository instanceof QuerySpecResourceRepository) {
+					resources = ((QuerySpecResourceRepository) resourceRepository).findAll(ids,
+							request.getQuerySpec(resourceInformation.getResourceClass()));
+				}
+				else {
+					resources = ((ResourceRepository) resourceRepository).findAll(ids, request.getQueryParams());
+				}
+				return getResponse(resourceRepository, resources, request);
+			}
+
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindIds(moduleRegistry.getResourceRegistry(),
+				queryAdapter, ids);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public <S extends T> JsonApiResponse update(S entity, QueryAdapter queryAdapter) {
+		return save(entity, queryAdapter, HttpMethod.PATCH);
+	}
+
+	public <S extends T> JsonApiResponse create(S entity, QueryAdapter queryAdapter) {
+		return save(entity, queryAdapter, HttpMethod.POST);
+	}
+
+	private <S extends T> JsonApiResponse save(S entity, QueryAdapter queryAdapter, HttpMethod method) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)